feat(header): Add component (2/3)

jpveooys opened the pull request:
This adds the header component, based on the GOV.UK equivalent. The original macro in the GOV.UK Frontend is only intended for use on GOV.UK, and isn't that customisable, so our own version has been added here.

I added some specific visual regression tests for hover states.

~~Unfortunately, there was a problem where the `window.matchMedia()` call by the JavaScript used in the header component was behaving inconsistently for the Pixel 5 project. For now, I've skipped the relevant tests for mobile Chromium, and will look to create a simpler repro for this to file a bug upstream. Edit: I created https://github.com/microsoft/playwright/issues/18919 regarding this.~~

jpveooys commented on src/nunjucks/moduk/components/header/macro.njk:
```diff
@@ -0,0 +1,75 @@
+
+{% macro modukHeader(params) %}
+
+{% set menuButtonText = params.menuButtonText if params.menuButtonText else 'Menu' %}
```
In GOV.UK Frontend the inner part of the macro is put in a separate file. I find that a bit annoying personally as I never know which file to look at.

jpveooys commented on examples-server/templates/example.njk:
```diff
@@ -5,6 +5,9 @@
   <link href="/css/index.css" rel="stylesheet">
 </head>
 <body>
+  <script>
+    document.body.className = ((document.body.className) ? document.body.className + ' js-enabled' : 'js-enabled');
+  </script>
```
This was missing: https://frontend.design-system.service.gov.uk/get-started/#5-get-the-javascript-working

jpveooys commented on e2e/visual-regression.spec.ts:
```diff
@@ -6,7 +6,13 @@ test.describe('@visual-regression', () => {
         test.describe(exampleName, () => {
-          test.beforeEach(async ({ page }) => {
+          test.beforeEach(async ({ page, browserName, isMobile }) => {
+            test.skip(
+              component === 'header' && exampleName === 'with-service-name-and-navigation'
+                && browserName === 'chromium' && Boolean(isMobile),
+              'Inconsistent window.matchMedia() behaviour in mobile Chromium',
+            )
```
See comment in PR description.

adamledwards commented:
Nice one a lot of work here it's a shame you had to rewrite the header. Let's hope that there are not many components like this.

jpveooys commented:
Have put this back to draft for the minute. We were missing `<meta name="viewport" content="width=device-width, initial-scale=1, viewport-fit=cover">` in our HTML templates for the examples server which was causing the Playwright mobile projects to render incorrectly.

Will do a separate PR to fix this as it'll change all the screenshots.

jpveooys commented:
I've rebased this on to #29 and updated and improved the tests. The skips have been removed as these are no longer required. Additional tests for mobile and with JavaScript disabled have also been added.

I've set up two new tags – `@desktop` and `@mobile` – to restrict tests to desktop or mobile. (`test.skip()` was an option, but the tags seemed like a cleaner way to do it.)
